### ISSUE TITLE
Update OpenStack/Shift Default New Project allocation & add the SU multiplier for OpenShift

### ISSUE THREAD

**Issue by joachimweyl** [nerc-project/coldfront-plugin-cloud#106]
This issue has 2 parts
1. Update the default values for the Allocation to be
   1. Both
      1. 1 CPU
      1. 4 GB RAM
      1. 0 GPU
   1. OpenStack only
      1. 1 Instance
      1. 2 Volumes
      1. 20 GB Volume Storage
      1. 0 GB Object Storage (S3)
   1. Replace/Set the text above the multiplier to be "Service Units (SU) of computing to allocate to the project. 1 CPU SU = xxx" 
      1. Where xxx is the values listed above
1. Add the multiplier to the OpenShift
   1. Make the OpenShift Request New Allocation work like the OpenStack Request New Allocation works in terms of having a cell you can enter a multiplier for the request.
here is an image of what it looks like in OpenStack 
![image](https://github.com/nerc-project/coldfront-plugin-cloud/assets/28782106/e1f69316-e5ef-4d1b-8b78-ca9b10342662)
and here is an image of what OpenShift looks like currently
![image](https://github.com/nerc-project/coldfront-plugin-cloud/assets/28782106/692b07e0-2d53-4633-89d3-c323d2a5550c)

* Do we want to add something along the lines of "S3 Storage and GPUs can be requested once your initial Allocation is approved" to the text above the multiplier?

**Comment by joachimweyl** [comment]
@knikolla I discussed this with @Milstein, and he was very clear that keeping the multiplier for both OpenStack/Shift was the way to go so that they can avoid as many extra modification requests, as that is currently a manual process.

**Comment by joachimweyl** [comment]
@jtriley I would like to confirm with you before moving forward with this change that you agree with the changes suggested above.

**Comment by joachimweyl** [comment]
approved in NERC Systems meeting 7/5.

**Comment by joachimweyl** [comment]
@naved001 is this something you feel you could do next sprint with @knikolla as backup if you have questions?

**Comment by naved001** [comment]
@joachimweyl maybe? depends on how much progress I make on the new ESI deployment.

**Comment by knikolla** [comment]
Done
![Screenshot 2023-10-18 at 1 18 20 PM](https://github.com/nerc-project/coldfront-plugin-cloud/assets/4123322/f73bd131-8b27-4968-819c-875ed7a542ad)

**Comment by joachimweyl** [comment]
Thank you @knikolla for this quick change.
The CPU should be 1 the RAM should be 4GB Ephemeral should be 20 GB.
@Milstein how many persistent volumes did you want here?
This issue also requests that we change OpenStack default values as well.

**Comment by knikolla** [comment]
If you want to change those values, the appropriate place is here https://github.com/nerc-project/coldfront-plugin-cloud/blob/d0b82e6c2a0a4f041872d0e5a372e133d0dd1a7d/src/coldfront_plugin_cloud/tasks.py#L31-L36

My label reflects the reality of what are default quotas at the moment.

**Comment by joachimweyl** [comment]
@Milstein  I found the OpenStack defaults in the same tasks.py file that housed the openshift information.
https://github.com/nerc-project/coldfront-plugin-cloud/blob/d0b82e6c2a0a4f041872d0e5a372e133d0dd1a7d/src/coldfront_plugin_cloud/tasks.py#L21-L29

**Comment by joachimweyl** [comment]
@knikolla does the STATIC_QUOTA overwrite the UNIT_QUOTA_MULTIPLIERS?
https://github.com/nerc-project/coldfront-plugin-cloud/blob/d0b82e6c2a0a4f041872d0e5a372e133d0dd1a7d/src/coldfront_plugin_cloud/tasks.py#L43-L48
if QUOTA_FLOATING_IPS was set to 1 in the UNIT_QUOTA_MULTIPLIERS and you only had 1 multiplier would you have 1 floating IP or 2 because of the STATIC_QUOTA being set to 2?

**Comment by knikolla** [comment]
https://github.com/nerc-project/coldfront-plugin-cloud/blob/d0b82e6c2a0a4f041872d0e5a372e133d0dd1a7d/src/coldfront_plugin_cloud/tasks.py#L40-L42

Quantity * Unit Quota Multiplier + Static Quota

> if QUOTA_FLOATING_IPS was set to 1 in the UNIT_QUOTA_MULTIPLIERS and you only had 1 multiplier would you have 1 floating IP or 2 because of the STATIC_QUOTA being set to 2?

1 * 1 + 2 = 3

**Comment by Milstein** [comment]
@joachimweyl : all current default values are reflected correctly as defined in this template.

Further adjustement needed:
For OpenStack --> Urgent Need Quota attribute to control/modify i. Security Groups. which is right now set default at 0 Security Groups/ project.

For OpenShift --> Urgent Need Quota attributes to control/modify for i. GPU and ii. Object Storage (NOBAA) --> These need to be part of our Cost Calculation.

**Comment by joachimweyl** [comment]
@Milstein current default values are not correct. Please see the list of defaults in the description of this issue.
As for the further Adjustments needed. That will require a new issue.

**Comment by knikolla** [comment]
> @joachimweyl : all current default values are reflected correctly as defined in this template.
> 
> Further adjustement needed: For OpenStack --> Urgent Need Quota attribute to control/modify i. Security Groups. which is right now set default at 0 Security Groups/ project.

It is currently set to default to 10 per project. ColdFront does not override that value.

```
openstack --os-cloud nerc quota show --default | grep secgroup 
| secgroup-rules        | 100                                                                                                                                                                           |
| secgroups             | 10                                                        
```

You can set a new default using the OpenStack commandline.

**Comment by Milstein** [comment]
@joachimweyl :  I think the following Changes to 1 Unit default valuese makes sense, please let me know your thoughts on these new values:
1. Update the default values for the Allocation to be
   1. Both
      1. 1 CPU
      1. 4 GB RAM
      1. 0 GPU
   1. OpenStack only
      1. 1 Instance
      1. 2 Volumes
      1. 20 GB Volume Storage
      1. 1 GB Object Storage (Swift)
   1. OpenShift only
      1. 2 Persistent Volume Claims(PVC)
      1. 5 GB Ephemeral Storage
      1. 20 GB Storage

**Comment by joachimweyl** [comment]
I will make changes to the tasks.py soon.

**Comment by joachimweyl** [comment]
@knikolla it appears that the ColdFront Request New Allocation still shows the old values.
I checked the current [code](https://github.com/nerc-project/coldfront-plugin-cloud/blob/main/src/coldfront_plugin_cloud/tasks.py#L20-L39) and it looks like it should be correct. Do we need to do anything after the merge to make this live?

![image](https://github.com/nerc-project/coldfront-plugin-cloud/assets/28782106/04c20e1b-d748-4453-9c64-0ca9fa8849c8)

![image](https://github.com/nerc-project/coldfront-plugin-cloud/assets/28782106/b835ab24-ae05-49c0-8c35-97f8e2f8c959)

**Comment by Milstein** [comment]
@knikolla : I wonder if that `quantity_label` text for this Resource Attribute is reverted back due to our recent version upgrade?

**Comment by Milstein** [comment]
I have updated the label as per our code.
